sonic_platform_base/chassis_base.py

```diff
@@ -95,16 +98,39 @@ def get_reboot_cause(self):
         """
         raise NotImplementedError
 
-    def get_component_versions(self):
+    def get_component_list(self):
+        """
+        Retrieves chassis components list such as BIOS, CPLD, FPGA, etc.
+
+        Returns:
+            A list containing component name.
+        """
+        return self._component_list
+
+    def get_firmware_version(self, component):
         """
         Retrieves platform-specific hardware/firmware versions for chassis
         componenets such as BIOS, CPLD, FPGA, etc.
+        Args:
+            component: A string, the component name.
 
         Returns:
             A string containing platform-specific component versions
         """
         raise NotImplementedError
 
+    def install_component_firmware(self, component, image_path):
```

Comment:
`def install_component_firmware(self, component_name, image_path):`, again, to go along with my above comments?